compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/NodeParser.java

```diff
@@ -147,4 +147,15 @@ public static ModulePartNode parseModulePart(String text) {
         BallerinaParser parser = ParserFactory.getParser(text);
         return parser.parse().createUnlinkedFacade();
     }
+
+    /**
+     * Parses the input as an intermediate clause.
+     *
+     * @param text the input
+     * @return a {@code IntermediateClauseNode}
+     */
+    public static IntermediateClauseNode parseIntermediateClause(String text) {
+        BallerinaParser parser = ParserFactory.getParser(text);
+        return parser.parseAsIntermediateClause(true, true).createUnlinkedFacade();
```

Comment:
@rdulmina and discussed the allowActions thing and conclude on giving it as an argument. 
```suggestion
    public static IntermediateClauseNode parseIntermediateClause(String text, boolean allowActions) {
        BallerinaParser parser = ParserFactory.getParser(text);
        return parser.parseAsIntermediateClause(allowActions).createUnlinkedFacade();
```